### PR TITLE
*: add send_response macro that helps interactive with grpc sinks

### PR DESCRIPTION
**Pull request by overvenus**
There are many gRPC sink types, some of them have `fail()` and `success()` methods and some of them only have `fail()`. It is difficult to write a function that handles all of the sinks, but macros can. In this PR, I add a new macro which handles most of the use cases and reduces about 150 lines of redundant code.

### CODE REVIEW

File: src/server/service/debug.rs [hunk 1]
```diff
@@ -198,16 +166,23 @@ impl<T: RaftStoreRouter + 'static + Send> debugpb_grpc::Debug for Service<T> {
...
     ) {
+        const TAG: &str = "scan_mvcc";
         let debugger = self.debugger.clone();
         let from = req.take_from_key();
         let to = req.take_to_key();
         let limit = req.get_limit();
         let future = future::result(debugger.scan_mvcc(&from, &to, limit))
-            .map_err(|e| error_to_grpc_error("scan_mvcc", e))
+            .map_err(|e| {
+                error!("{} failed: {:?}", TAG, e);
```
Comment (siddontang) [inline src/server/service/debug.rs]: why use error here?

**Comment by overvenus** [comment]
I talked with @BusyJay offline, we think we can provide `result` functions for every sink types in the grpcio, so I am going to close this PR.

**Pull request closed without merging**